vtrace/platforms/gdbstub.py

```diff
@@ -2671,9 +2711,8 @@ def _handleStepi(self, sig=signal.SIGTRAP):
         """
         # After every step the current signal information should be provided (it 
         # will likely be SIGTRAP)
-        sig = self._serverStepi(sig)
-        res = b'S' + self._encodeGDBVal(sig)
-        return res
+        self._serverStepi(sig)
+        return None
```

Comment:
are we making `_serverStepi()` responsible for the response now?

Reply:
That is the way I have it structured at the moment? I'm not sure if this is a good idea or not.

I did this way because first I looked at the "continue" command because it shouldn't send a message until the emulator halts.

So I was going to make "stepi" work in a similar way. it does happen to be that the way the default base server implements `_serverStepi` means that the "stopped state" of the emulator is determined in the GDB thread but depending on how the emulator is implemented I wasn't sure if this would always work?

I can return stepi to returning the reply msg because for the moment that does work. if we don't need a change yet it probably should be the same as it was, right?

Reply:
or does STEPI produce a start and stop differently?

i'd love to document the expected flow here somewhere.  perhaps here, since this is where it departs from other command sequences?

Reply:
stepi is essentially the same, "current state of the system" reply should be received when the emulator is stopped. but unless it's implemented asynchronously like breakpoints it doesn't _have_ to have separated "run" and "report stop" events.

I was about to implement it using breakpoints until I remembered that if you `stepi` over a conditional branch instruction that things wouldn't work as desired.

Reply:
gotcha.  yeah, i would put it back the way it was, if that still works.  unless there's a good reason, i'd like to leave the paradigm as coherent as possible.

thanks!

Reply:
will do :+1:

Reply:
the alternative, which would probably work for the PPC Emulator but perhaps cause issues elsewhere, is to queue a special Exception (kinda like what does a HALT), which would fire the Stepi Response code.

at the very least, we *can* have _serverStepi() be responsible for triggering the response.  the reason i'm thinking that is a benefit is this:  what do we do if the single instruction causes an exception?

i apologize, i know this is a reversal in my stance.  what are your thoughts?

Reply:
I thought about that, but if that's the case then the emulator can override `_handleStepi`.  I don't think we need to change the structure of the vivisect gdbstub.

Reply:
we can keep this as a thought for the future...
for now, i think we're good.